### DISCUSSION TITLE
Rapid Blink vs Slow Blink

### DISCUSSION THREAD

**Discussion by Hi-ImKyle** [spectreconsole/spectre.console#1557] [Q&A] [answered]
Consider the following lines of code.

```csharp
AnsiConsole.MarkupLine("[slowblink]I am slow blinking[/]");
AnsiConsole.MarkupLine("[rapidblink]I am rapid blinking[/]");
```

Without running it, which do you think blinks faster?

Spoiler. They both blink at the same speed.

So why is it called rapid blink? Why does the slow blink state it blinks "slowly" on the docs? Is the slow blink rapid or is the rapid blink slow?

But real talk, is there a reason why they are the same speed? Is this a incompatibility thing? I have tried both Windows Terminal and CMD. Neither of which seem to change the blinking speed. Is this worthy of a new issue? I decided against making it a issue for the sake of just asking first. Maybe there is some information I am missing that helps me understand why they are the same speed.

**Comment by patriksvensson** [comment, marked as answer]
This depends on your terminal, and if the terminal supports/differentiates between rapid  and slow blink.

**Reply by patriksvensson** [reply]
To clarify; we send VT-100 escape codes to the terminal, but how the terminal interprets them is up to it. I suspect Windows Terminal and cmd.exe do not support rapid blinking and just fall down to slow blinking.

Here is an example of how different it can look between 3 different terminals on macOS:

https://github.com/spectreconsole/spectre.console/assets/357872/06f48c9b-fd97-403b-98fd-7a39c935d0ed

**Reply by Hi-ImKyle** [reply]
Seems it's just windows it affects then, not sure of a terminal on windows this works on. Fair enough, thanks.
